google_benchmark/include/benchmark/benchmark.h

```diff
@@ -1582,27 +1585,67 @@ class Fixture : public internal::Benchmark {
+      this->SetName(CUR_FILE + ns_##BaseClass##_##Method +    \
+                    #BaseClass NAME_START #Method NAME_END);  \
     }                                                         \
                                                               \
    protected:                                                 \
     void BenchmarkCase(::benchmark::State&) override;         \
   };
 
+#ifdef CODSPEED_ENABLED
 #define BENCHMARK_TEMPLATE1_PRIVATE_DECLARE_F(BaseClass, Method, a) \
+  STATIC_NAMESPACE_STRING(ns_##BaseClass##_##Method);               \
   class BaseClass##_##Method##_Benchmark : public BaseClass<a> {    \
    public:                                                          \
     BaseClass##_##Method##_Benchmark() {                            \
-      this->SetName(#BaseClass "<" #a ">/" #Method);                \
+      this->SetName(CUR_FILE + ns_##BaseClass##_##Method +          \
+                    #BaseClass "[" #Method ", " #a "]");            \
     }                                                               \
                                                                     \
    protected:                                                       \
     void BenchmarkCase(::benchmark::State&) override;               \
   };
+#define BENCHMARK_TEMPLATE2_PRIVATE_DECLARE_F(BaseClass, Method, a, b) \
+  STATIC_NAMESPACE_STRING(ns_##BaseClass##_##Method);                  \
+  class BaseClass##_##Method##_Benchmark : public BaseClass<a, b> {    \
+   public:                                                             \
+    BaseClass##_##Method##_Benchmark() {                               \
+      this->SetName(CUR_FILE + ns_##BaseClass##_##Method +             \
+                    #BaseClass "[" #Method ", " #a ", " #b "]");       \
+    }                                                                  \
+                                                                       \
+   protected:                                                          \
+    void BenchmarkCase(::benchmark::State&) override;                  \
+  };
 
+#define BENCHMARK_TEMPLATE_PRIVATE_DECLARE_F(BaseClass, Method, ...)       \
+  STATIC_NAMESPACE_STRING(ns_##BaseClass##_##Method);                      \
+  class BaseClass##_##Method##_Benchmark : public BaseClass<__VA_ARGS__> { \
+   public:                                                                 \
+    BaseClass##_##Method##_Benchmark() {                                   \
+      this->SetName(CUR_FILE + ns_##BaseClass##_##Method +                 \
+                    #BaseClass "[" #Method ", " #__VA_ARGS__ "]");         \
+    }                                                                      \
+                                                                           \
+   protected:                                                              \
+    void BenchmarkCase(::benchmark::State&) override;                      \
+  };
+#else
```

Comment:
We could add a comment here to express that it's the CODSPEED_ENABLED if, makes it a bit more readable